fastlane/swift/LaneFileProtocol.swift

```diff
@@ -17,20 +17,20 @@ public protocol LaneFileProtocol: class {
     func recordLaneDescriptions()
     func beforeAll(with lane: String)
     func afterAll(with lane: String)
-    func onError(currentLane: String, errorInfo: String)
+    func onError(currentLane: String, errorInfo: String, errorClass: String?, errorMessage: String?)
 }
 
 public extension LaneFileProtocol {
     var fastlaneVersion: String { return "" } // Defaults to "" because that means any is fine
     func beforeAll(with _: String) {} // No-op by default
     func afterAll(with _: String) {} // No-op by default
-    func onError(currentLane _: String, errorInfo _: String) {} // No-op by default
```

Comment:
Why this default public implementation was deleted?

Reply:
It is now living in the `LaneFile` superclass which it will record the called lanes to avoid `afterAll` being called unnecessarily. Also, for some reason even if the Fastfile overloaded the protocol's implementation, it gets called anyway and ignoring the new implementation (couldn't figure out why exactly).

I'll address the changes you mention once I test that they're working properly, thanks for the feedback! ❤️❤️